internal/services/mobilenetwork/mobile_network_sim_group_resource.go

```diff
@@ -0,0 +1,281 @@
+package mobilenetwork
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/identity"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/location"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/mobilenetwork/2022-11-01/mobilenetwork"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/mobilenetwork/2022-11-01/simgroup"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+type SimGroupModel struct {
+	Name             string                       `tfschema:"name"`
+	EncryptionKeyUrl string                       `tfschema:"encryption_key_url"`
+	Identity         []identity.ModelUserAssigned `tfschema:"identity"`
+	Location         string                       `tfschema:"location"`
+	MobileNetworkId  string                       `tfschema:"mobile_network_id"`
+	Tags             map[string]string            `tfschema:"tags"`
+}
+
+type SimGroupResource struct{}
+
+var _ sdk.ResourceWithUpdate = SimGroupResource{}
+
+func (r SimGroupResource) ResourceType() string {
+	return "azurerm_mobile_network_sim_group"
+}
+
+func (r SimGroupResource) ModelObject() interface{} {
+	return &SimGroupModel{}
+}
+
+func (r SimGroupResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return simgroup.ValidateSimGroupID
+}
+
+func (r SimGroupResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"mobile_network_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: mobilenetwork.ValidateMobileNetworkID,
+		},
+
+		"encryption_key_url": { // needs UserAssignedIdentity
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ValidateFunc: validation.IsURLWithHTTPorHTTPS,
+		},
+
+		"identity": commonschema.SystemOrUserAssignedIdentityOptional(),
```

Comment:
the documentation states that only `UserAssigned` is supported.. but this doesn't match - @ziyeqf this'll want updating?

Reply:
oh, submitted a pr #20474 to update this.